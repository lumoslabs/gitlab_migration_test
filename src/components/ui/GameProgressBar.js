
import React from 'react';
import ProgressBar from 'react-bootstrap/ProgressBar';
import { css, StyleSheet } from 'aphrodite/no-important';
import commonStyles from '../../styles/commonStyles';
import { commonNestStylesWithProps } from '../../styles/commonNestStyles';

import './cocos.css';

function GameProgressBar(props) {
  return (
    <div className = { css([commonStyles.flexColumnAllCenter, commonStyles.flexJustifyCenter,
      commonNestStylesWithProps({ clientHeight: window.usableClientHeight}).fullHeight,
      styles.progressBarDiv
    ])}>
      <ProgressBar className = { css([styles.progressOverride]) } animated={ false } now={props.progressLevel} />
      <p className = { css(styles.loadingText) }>
        Loading { props.name }
      </p>
    </div>
  );
}

const styles = StyleSheet.create({
  progressBarDiv: {
    position: 'absolute',
  },
  progressOverride: {
    width: '64vmin',
    height: '3vmin',
    borderRadius: '1.5vmin',
  },
  loadingText: {
    color: 'rgb(0, 0, 0)',
    fontSize: '3vmin',
    fontFamily: 'MuseoSans500',
    marginTop: '2.5vmin',
  }
});

export default GameProgressBar;
