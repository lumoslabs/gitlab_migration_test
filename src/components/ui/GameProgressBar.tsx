import React from 'react';
import ProgressBar from 'react-bootstrap/ProgressBar';
import { css, StyleSheet } from 'aphrodite/no-important';
import commonStyles from '../../styles/commonStyles';

export interface IGameProgressBarProps {
  name: string;
  progressLevel: number;
}

const GameProgressBar = (props: IGameProgressBarProps): JSX.Element => {
  return (
    <div className={css([commonStyles.flexColumnAllCenter, commonStyles.flexJustifyCenter, styles.progressBarDiv])}>
      <ProgressBar className={css(styles.progressOverride)} animated={false} now={props.progressLevel} />
      <p className={css(styles.loadingText)}>
        Loading {props.name}
      </p>
    </div>
  );
}

const styles = StyleSheet.create({
  progressBarDiv: {
    position: 'absolute'
  },
  progressOverride: {
    width: '64vmin',
    height: '3vmin',
    borderRadius: '1.5vmin'
  },
  loadingText: {
    color: 'rgb(0, 0, 0)',
    fontSize: '3vmin',
    fontFamily: 'MuseoSans500',
    marginTop: '2.5vmin'
  }
});

export default GameProgressBar;
