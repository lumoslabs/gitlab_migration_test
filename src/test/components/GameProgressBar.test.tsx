import React from 'react'
import GameProgressBar from '@components/ui/GameProgressBar'
import { render } from '@testing-library/react'

it('matches snapshot', () => {
  const { asFragment } = render(
    <GameProgressBar
      progressLevel={50}
   />, {}
  )
  expect(asFragment()).toMatchSnapshot()
})