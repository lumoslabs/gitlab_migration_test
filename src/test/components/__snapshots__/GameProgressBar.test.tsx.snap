// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flexColumnAllCenter_1vexgx1-o_O-flexJustifyCenter_jfsfb5-o_O-progressBarDiv_xge2iy"
  >
    <div
      class="progressOverride_1kjv81f progress"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="50"
        class="progress-bar bg-lumos"
        role="progressbar"
        style="width: 50%;"
      />
    </div>
    <p
      class="loadingText_u0pnee"
    >
      Loading
    </p>
  </div>
</DocumentFragment>
`;
